---
title: "R Notebook"
output: html_notebook
---
data from https://rawgit.com/bioinformatics-core-shared-training/microarray-analysis/master/affymetrix.nb.html


LOAD LIBRARIES
```{r}
library(oligo)
library(GEOquery)
library(limma)
library (RColorBrewer)
library(cluster)
library(biomaRt)
library(purrr)
```

READ INPUT
ADD INFO

```{r}
targetsFile <- "phenodata_OK.txt"
pd <- read.AnnotatedDataFrame(targetsFile,header=TRUE,sep="",row.names=1)
# pData(pd)
rawData <- read.celfiles(rownames(pData(pd)))
filename <- sampleNames(rawData)
pData(rawData)$filename <- filename
sampleNames <- sub(".*_", "", filename)
sampleNames <- sub("-HuGene.CEL.gz$", "", sampleNames)
sampleNames(rawData) <- sampleNames
pData(rawData)$group <- ifelse(grepl("^IDC|^[PS]", sampleNames(rawData)),
                               "CASE", "Control")
pData(rawData)
```

QC ANALYSIS: plot raw array data

```{r}
# par(mfrow=c(2,2))
# layout(matrix(c(1,2,3,4), 2, 2, byrow = TRUE))
image(rawData[,1])
image(rawData[,2])
image(rawData[,3])
image(rawData[,4])
```
NORMALIZE reading data
```{r}
normData <- rma(rawData)
```

COMPARE RAW WITH NORM DATA

MAPLOT RAW
```{r}
MAplot(rawData[, 1:4], pairs=TRUE)
```
MAPLOT NORM
```{r}
MAplot(normData[, 1:4], pairs=TRUE)
```
BOXPLOT AND HIST RAW VS NORM
```{r}
par(mfrow=c(1,2))
boxplot(rawData,target='core',main="raw")
boxplot(normData,main="norm")
```
```{r}
hist(rawData,target="core")
hist(normData)
```

LOAD PREPROCESSED DATA FROM GEO TO REPRODUCE RESULTS
SERIES MATRIX LOADING...

```{r}
preprocessedData <- getGEO(filename="GSE67492_series_matrix.txt")
gsms <- "XX1100"
sml <- c()
for (i in 1:nchar(gsms)) { sml[i] <- substr(gsms,i,i) }
sel <- which(sml != "X")
preprocessedData <- preprocessedData[ , sel]
```

CHECK THAT RESULTS LOOK LIKE NORMALIZED DATA FROM CEL FILES
```{r}
boxplot(preprocessedData)
MAplot(preprocessedData[, 1:4], pairs=TRUE)
hist(preprocessedData)
```

Continue with downstream analysis (genes and pathways)

```{r}
varFiltered <- varFilter (preprocessedData)
# dim (varFiltered)
# nrow (preprocessedData) / nrow (varFiltered)
euc.dist <- dist (t(exprs(varFiltered)))
#euc.dist
corMat <- cor(exprs(varFiltered))
#corMat
cor.dist <- as.dist(1 - corMat)
#cor.dist
clust <- hclust(cor.dist)
# clust
# names(clust)
# clust$order
# plot(clust)
clust.ward <- hclust(cor.dist,method = "ward.D")
# par(mfrow=c(1,2))
# plot(clust)
# plot(clust.ward)
# plot(clust,labels=paste("Sample", 1:4))
pd <- pData(preprocessedData)
# pd
#View(pd)
plot(clust,labels=pd$description)
```

```{r}
# library(WGCNA)
# SampleGroup <- c("sample1","sample2","ctrl1","ctrl2")
# groupColours <- rep("yellow",length(SampleGroup))
# groupColours[3:4] <- "blue"
# # groupColours
# clust.euclid = hclust(euc.dist)
# plotDendroAndColors(clust.euclid,colors=groupColours)
```

```{r}
geneVar = rowSds(exprs(preprocessedData))
# sd(exprs(preprocessedData)[1,])
# geneVar[1]
# sd(exprs(preprocessedData)[2,])
# geneVar[2]
# length(geneVar)
highVarGenes = order (geneVar, decreasing = TRUE )[1:100]
SampleGroup <- c("sample1","sample2","ctrl1","ctrl2")
groupColours <- rep("yellow",length(SampleGroup))
groupColours[3:4] <- "blue"

# heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, ]),labCol = SampleGroup)
# heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, order(SampleGroup)]),
#          labCol = SampleGroup,
#          ColSideColors = as.character(groupColours))
# heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, order(SampleGroup)]),
#          labCol = SampleGroup[order(SampleGroup)],
#          ColSideColors = as.character(groupColours))
# heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, ]),
#          labCol = SampleGroup , Colv=NA)
heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, order(SampleGroup)]),
         labCol = SampleGroup[order(SampleGroup)], Colv = NA)
```

```{r}
# display.brewer.all()
hmcol <- brewer.pal(11 , "RdBu")
heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, order(SampleGroup)]) , 
         labCol = SampleGroup[order(SampleGroup)],
         col=hmcol)


# heatmap (as.matrix(exprs(preprocessedData)[highVarGenes, order(SampleGroup)]),
#          labCol = SampleGroup[order(SampleGroup)],
#          ColSideColors = as.character(groupColours), col=hmcol)
```

```{r}
 supervised.clus <- pam(euc.dist,k=2)
 clusplot(supervised.clus)
#supervised.clus$clustering
```
 


```{r}
pca.geneExpression <- prcomp(t(exprs(varFiltered)))
summary(pca.geneExpression)
# plot(pca.geneExpression)
# plot(pca.geneExpression$x[,1],pca.geneExpression$x[,2])
# plot(pca.geneExpression$x[,1],pca.geneExpression$x[,2],
#      pch=16,col=as.character(groupColours))
plot(pca.geneExpression$x[,1],pca.geneExpression$x[,2],
     pch=16,col=as.character(groupColours))
legend("bottomright",fill=c("yellow","blue"),legend=c("case","control"))
text(pca.geneExpression$x[,1],pca.geneExpression$x[,2]-0.01,labels = pd$geo_accession)
boxplot(pca.geneExpression$x[,1] ~ SampleGroup)

```

```{r}
pData(preprocessedData)$source_name_ch1[4] <- pData(preprocessedData)$source_name_ch1[3]
design <- model.matrix(~0+droplevels(pData(preprocessedData)$source_name_ch1))
# design
colnames(design) <- c("CONTROL","CASE")
design
```
```{r}
fit1<-lmFit(preprocessedData, design)
contrasts <- makeContrasts(CONTROL - CASE, levels=design)
fit2 <- contrasts.fit(fit1, contrasts)
fit3 <- eBayes(fit2)
topTable(fit3)
```

```{r}
tab<-topTable(fit3, number = 3e6, adjust.method = "BH", sort.by = "p")
pr<-row.names(tab)
x <- hugene10sttranscriptclusterENSEMBL
mapped_genes <- mappedkeys(x)
xx <- as.list(x[mapped_genes])
sy <- xx[pr]
head(sy)
length(labels(sy)[!is.na(labels(sy))])
```
```{r}

mart <- useMart("ensembl")
ensemblHuman <- useDataset ("hsapiens_gene_ensembl",mart=mart)
```

```{r}
myGenes <- as.character(compact(sapply(sy, function(k){as.character(k[1])})))
myGenesDF <- as.data.frame(myGenes)
colnames(myGenesDF) <- 'ensembl_gene_id'
myAttr <- c('ensembl_gene_id', 'description', 'external_gene_name', 'chromosome_name')

myGenesRes <- getBM(attributes = myAttr, filters='ensembl_gene_id', values=myGenesDF, mart=ensemblHuman)
head(myGenesRes)
```

```{r}
# sy.probes <- labels(sy)[as.character(sapply(sy, function(k){as.character(k[1])})) != "character(0)"]
# myGenesDF$symbol <- myGenesRes[match(myGenesDF[,1],myGenesRes[,1]),3]
# myGenesDF$probeid <- sy.probes
# matchtab <- tab[match(myGenesDF[,3],row.names(tab)),]
# matchtab$probeid <- row.names(matchtab)
# myGenesDF.stats <- merge(myGenesDF,matchtab,by="probeid",sort=F)
# head(myGenesDF.stats)
# dim(myGenesDF.stats)
```

```{r}
xEN <- hugene10sttranscriptclusterENTREZID
mapped_genesEN <- mappedkeys(xEN)
xxEN <- as.list(xEN[mapped_genesEN])
syEN <- xxEN[pr]
head(syEN)
length(labels(syEN)[!is.na(labels(syEN))])
syEN.probes <- labels(syEN)[as.character(sapply(syEN, function(k){as.character(k[1])})) != "character(0)"]
syEN.entrez <- as.character(compact(sapply(syEN, function(k){as.character(k[1])})))
length(syEN.probes)
length(syEN.entrez)
syEN.entrez.df <- as.data.frame(syEN.probes) 
colnames(syEN.entrez.df) <- 'probeid'
syEN.entrez.df$entrezid <- syEN.entrez
head(syEN.entrez.df)
```
```{r}
sy.probes <- labels(sy)[as.character(sapply(sy, function(k){as.character(k[1])})) != "character(0)"]
myGenesDF$symbol <- myGenesRes[match(myGenesDF[,1],myGenesRes[,1]),3]
myGenesDF$probeid <- sy.probes
myGenesDF <- merge(myGenesDF,syEN.entrez.df,by="probeid",sort=F)

matchtab <- tab[match(myGenesDF[,1],row.names(tab)),]
matchtab$probeid <- row.names(matchtab)
myGenesDF.stats <- merge(myGenesDF,matchtab,by="probeid",sort=F)
head(myGenesDF.stats)
dim(myGenesDF.stats)
```
PATHWAY ANALYSIS

```{r}
library(ReactomePA)
geneList <- myGenesDF.stats$entrezid
pathway1 <- enrichPathway(geneList)
head(as.data.frame(pathway1))
dotplot(pathway1)
barplot(pathway1,showCategory=8)
emapplot(pathway1)
cnetplot(pathway1, categorySize="pvalue", foldChange=geneList)
```
```{r}
y <- gsePathway(geneList, nPerm=10000,
                pvalueCutoff=0.2,
                pAdjustMethod="BH", verbose=FALSE)
res <- as.data.frame(y)
head(res)
emapplot(y, color="pvalue")
gseaplot(y, geneSetID = "R-HSA-69242")
```


GO ANALYSIS
https://rawgit.com/bioinformatics-core-shared-training/microarray-analysis/master/enrichment.nb.html
```{r}
colonData <- getGEO(filename="GSE67492_series_matrix.txt")
gsms <- "XX1100"
sml <- c()
for (i in 1:nchar(gsms)) { sml[i] <- substr(gsms,i,i) }
sel <- which(sml != "X")
colonData <- colonData[ , sel]
annotation(colonData) <- "pd.hugene.1.0.st.v1"
colonData
filt <- nsFilter(colonData)
```

