---
title: "Muscle RNASeq - Brian Drew"
author: "[Baker Bioinformatics (Artika Nath)](https://intranet.baker.edu.au/research-resources/platform-technologies/bioinformatics-program/)"
date: "`r format(Sys.time(), '%d %B %Y')`"
mail: "bioinformatics@baker.edu.au"
output:
  epuRate::BAKER:
    toc: TRUE
    number_sections: TRUE
    code_folding: "hide"
---
  
  
<script>
$(document).ready(function() {
  $items = $('div#TOC li');
  $items.each(function(idx) {
    num_ul = $(this).parentsUntil('#TOC').length;
    $(this).css({'text-indent': num_ul * 10, 'padding-left': 0});
  });
});
</script>

div.update {
  min-height: 50px;
}

***

# Load libraries

```{r warning=FALSE, message = FALSE }
library(ggplot2)
library(dplyr)
library(stringr)
library(readxl)
library(rmarkdown)   
library(tidyr)
library(epuRate) 
library(dplyr)
library(png)
library(gridExtra)
library(limma)
library(reshape)
library(RColorBrewer)
library(RUVSeq)
library(kableExtra)
library(DESeq2)
library(gplots)
library(fgsea)
library(msigdbr)
library(org.Mm.eg.db)
library(circlize)
library(ComplexHeatmap)
library(VennDiagram)
library(ggrepel)
library(pheatmap)
```


# Study Design 

:::infonote
<h4 class="alert-heading"> **Study Design**</h4>
<p> RNA-seq data of 12 muscle tissues from mouse model were generated </p>
<p> Sequencing was done on Nov-Seq6000 platform - run on two lanes </p>
<p> 66 BP Paired-end reads were generated </p>
:::

## Sample Info

:::infonote
  <h4 class="alert-heading"> **Raw Read Depth Summary**</h4>
  <p> Read Depth Range =  38.67 - 75.40 million reads </p>
  <p> Median Read Depth =  51.7 Million Read</p>
  <p> </p>
:::

```{r warning=FALSE, message = FALSE, fig.height = 2, fig.width = 10}
dat <- read.csv("InputFiles/ReadSummaryStat_RawFastq_MouseMuscle.csv")
opts <- dat[,c("Geno", "Mouse", "PercentOptical")]
dat1 <- dat[,c("Geno", "Mouse", "ReadsMillion")]
dat2 <- dat1
dat2 <- dat2[order(dat2$Geno),]
dat2$ReadsMillion <- round(dat2$ReadsMillion, 2)
  dat2 %>% 
  group_by(Geno) %>%
  kbl(caption = "Distribution of Raw Read Depth Across Muscle Samples") %>%
  kable_classic(full_width = F, html_font = "Cambria") %>%
  group_rows("Group A", 1, 6) %>%
  group_rows("Group D", 7, 12)
```

# PreAlign QC

## Merge Reads

:::infonote
<h4 class="alert-heading"> **Step1-Merge Reads**</h4>
For each sample, sequencing was done across two lanes  
Hence, lane-level reads were merged to sample level reads prior to any downstream analysis
:::


## FastQC

:::infonote
<h4 class="alert-heading"> **Step2-FASTQC**</h4>
  <p> The qualities of the raw sequence reads were assessed using FastQC. </p>
  <p> Based on the quality reports, read trimming was not required (Phred score > 35).  </p>
 <p> </p>
:::


## Remove Opticals

:::infonote
<h4 class="alert-heading"> **Step3-Remove Optical duplicates**</h4>
Optical duplicates were removed using BBMap clumpify (https://github.com/BioInfoTools/BBMap)
BBMap version 38.75

**What are optical duplicates** - generated by the same DNA cluster on a sequencing flow cell but are mistakenly identified as separate clusters by image capture software used during sequencing.

For patterned flow cell, its recommended that duplicates separated by a distance less than 2500 pixels is removed.

Optical duplicates were removed prior to alignment
:::

**Figure A** - Shows the distribution of Raw Reads after removing optical duplicates. After removing optical Duplicates the read depth range is between 34.33 - 65.62 million reads; Median = 45.2 million reads.   

**Figure B** - Shows the distribution of the percentage of optical duplicates in the Muscle samples. 
Both the figures are stratified by "Geno" groups

```{r warning=FALSE, message = FALSE, fig.height = 5, fig.width = 10}
P <- ggplot(dat, aes(x=Mouse, y=MillionAfterRemovingOpticals, fill=Geno)) +
geom_bar(stat="identity", width=.65, show.legend = FALSE) + facet_grid(~ Geno, scale="free") +
 theme (axis.text.x  = element_text(size=12, angle = 45, vjust = 0.36, hjust=0.36), axis.text.y = element_text(size=12)) + theme(axis.title=element_text(size=12,face="bold")) + 
 theme(strip.text.x = element_text(size = 16, colour = "black")) + ylab("Million Reads (After removing opticals)") + 
  ggtitle("A - Read Depth Distribution After Removing Opticals")

Q<-ggplot(dat, aes(x=Mouse, y=PercentOptical, group=Geno)) +
  geom_line(aes(color=Geno), show.legend = FALSE)+ 
  geom_point(aes(color=Geno), show.legend = FALSE) + facet_grid(~ Geno, scale="free") + 
  theme (axis.text.x  = element_text(size=12, angle = 45, vjust = 0.36, hjust=0.36), axis.text.y = element_text(size=12)) + theme(axis.title=element_text(size=12,face="bold")) + 
 theme(strip.text.x = element_text(size = 16, colour = "black")) + ylab("% Optical Duplicates Per Sample") + 
  ggtitle("B-Distribution of Optical Duplicates (Percentages)")

grid.arrange(P, Q, nrow = 1)
```

# PostAlign QC

## Alignment

:::infonote
  <h4 class="alert-heading"> **Step4-STAR Alignment**</h4>
  
The processed reads were aligned to mm10 version of the mouse genome using the STAR aligner

82.4-87.9% (median = 85%) of the reads were uniquely mapped to the mm10 genome. 

:::

```{r warning=FALSE, message = FALSE, fig.height = 5, fig.width = 10}
star <- read.csv("InputFiles/SummaryStat_StarAlignment_MouseMuscle.csv")
sub <- c("Mouse", "TotUniquelyMappedMillion", "TotMultiMappedMillion", "TotUnmappedMillion")
a <- melt(star[, which(colnames(star) %in% sub)])
sampleInfo1 <- star[,c("names", "Geno", "Mouse")]
a1 <- merge(a, sampleInfo1, by.x = "Mouse", by.y = "Mouse")
a2 <- a1[order(a1$Geno, a1$names),]

P <- ggplot(a2, aes(x=Mouse, y=value, fill=(factor(variable, levels=c("TotUnmappedMillion", "TotMultiMappedMillion", "TotUniquelyMappedMillion"))))) + geom_bar( stat="identity", width=.65,  color="gray17") + guides(fill=guide_legend(title="Mapping Stats", reverse = TRUE)) + facet_grid(~ Geno, scale="free") + facet_grid(~ Geno, scale="free") + 
  theme(axis.text.x  = element_text(size=14, angle = 45, vjust = 0.36, hjust=0.36, margin = margin(r = 0)), axis.text.y = element_text(size=14, margin = margin(r = 0))) + theme(axis.title=element_text(size=12,face="bold")) + theme(strip.text.x = element_text(size = 16, colour = "black")) + ylab("Reads in Millions") + 
ggtitle("Alignment summary (mm10 reference) of the processed reads,by Geno Group")
P + scale_fill_manual(values=c("grey", "yellow", "#F8766D"), labels=c("Un-mapped", "Multi-mapped", "Uniquely-mapped")) 
```

## Getting Counts

:::infonote
<h4 class="alert-heading"> **Step5-Obtaining Counts**</h4>

Reads mapped to genes were counted using the featureCounts tools
59.5 - 75.6% of the reads were assigned to genic features by featureCounts tool. 
:::

```{r warning=FALSE, message = FALSE, fig.height = 5, fig.width = 10}
a2 <- read.csv("InputFiles/FeatureCounts_SummaryStatsMuscle.csv")
a3 <- a2[,which(colnames(a2) %in% c("Mouse", "Geno", "Assigned_Per", "Unassigned_MultiMapping_Per", "Unassigned_NoFeatures_Per",  "Unassigned_Ambiguity_Per"))]
b <- melt(a3)
xlabel <- "Mouse Liver Sample"
P <- ggplot(b, aes(x=Mouse, y=value, fill=(factor(variable, levels=c("Unassigned_Ambiguity_Per", "Unassigned_NoFeatures_Per", "Unassigned_MultiMapping_Per","Assigned_Per"))))) + 
  geom_bar( stat="identity", width=.65,  color="gray17") + guides(fill=guide_legend(title="Feature Counts Stats", reverse = FALSE)) +scale_y_continuous(breaks=seq(0,100,10)) +  ylab("Percentage") + xlab(xlabel) + 
  theme(axis.text.x = element_text(face="bold", color="black", size=12, angle = 45, hjust=1), axis.text.y = element_text(color="black", size=12), axis.title.x = element_text(color="black", size=14, face="bold"), axis.title.y = element_text(face="bold", size=14),legend.title=element_text(size=10), legend.text=element_text(size=9)) + ggtitle("Summary of the reads mapped to Genic Feautures") + coord_flip()
P + scale_fill_manual(values=c("#00BFC4", "#C77CFF", "#7CAE00", "#F8766D"), labels=c("% Unassigned_Ambiguity", "% Unassigned_NoFeatures", "% Unassigned_MultiMapping","%Assigned"))
```

# Counts(Sanity Checks)

:::infonote
<h4 class="alert-heading"> **Sanity Checks of Count Data**</h4>
Lowly expressed genes were removed using a threshold of > 1 CPM in at least 9 samples (genes that had CPM > 1 on at least 3 samples were retained)

Based on the above criterion, N=11,917  genes are retained

Trimmed Mean ofÂ M-values normalization (TMM) was done to account for library size variation between samples. 
:::

```{r warning=FALSE, message = FALSE, fig.height = 5, fig.width = 10}
#setwd("/Users/anath/Dropbox/Artika/Projects/BakerBioinformatics/BrainDrew/BrianDrew_Oct2021/")
counts <- read.table("InputFiles/MuscleMouse_counts.txt",check.names = FALSE, header=TRUE)
info <- read.csv("InputFiles/sample_info.csv")
colnames(counts) <- gsub("_S.*.bamAligned.sortedByCoord.out.bam", "", colnames(counts))
colnames(counts)[7:18] <- paste0("M", colnames(counts)[7:18])
counts_sub <- counts[,7:18]
info_sub <- info[which(info$Mouse %in% colnames(counts_sub)),]
info_sub <- info_sub[order(info_sub$Geno),]
info_sub$Mouse <- as.character(info_sub$Mouse)
counts_sub1 <- counts_sub[,info_sub$Mouse]
#table(colnames(counts_sub1) == info_sub$Mouse)
group <- info_sub$Geno
x <- edgeR::DGEList(counts=counts_sub1, genes=counts[,c(1,6)], samples = info_sub, group=group)
keep <- rowSums(cpm(x) >1) > 3
xkeep <- x[keep, , keep.lib.sizes = FALSE]
xkeep$samples$lib.size <- colSums(xkeep$counts)
```

## Boxplot Samples
Boxplots before(A) and after(B) TMM normalisation. The boxplot shows the distribution of gene expression (N-11,683) levels of samples.Samples are coloured by their respective "Geno" group

```{r, echo = F, fold.plot=FALSE, fig.width=10,fig.height=5}
#TMM normalisation
d <- calcNormFactors(xkeep, method = "TMM")
TMM_normalised_log2 <- cpm(d, log = TRUE)
TMM_normalised_log2_1 <- cbind(d$genes, TMM_normalised_log2)
#d$samples$norm.factors
dat <-d$counts
datNorm <- TMM_normalised_log2
color <- c(rep("blue", 6), rep("green", 6))

boxplot(log2(dat + 1), col = color, las=2, outline=FALSE,ylab="log2 Raw Counts",  main="A. Raw counts (log2 transformed)")
legend("topright", inset=0.001, title="Group",c("GenoA", "GenoD"), 
fill= unique(color), cex=0.8, ncol = 3)

boxplot(datNorm, col = color, las=2, outline=FALSE,ylab="B. TMM normalised CPM values (log2 transformed)", 
main="TMM Normalised counts")
legend("topright", inset=0.001, title="Group",c("GenoA", "GenoD"), 
fill= unique(color), cex=0.8, ncol = 3)
```

## PCA Analysis

```{r, fold.output=FALSE, fold.plot=FALSE}
PCA.data_T <- prcomp(t(TMM_normalised_log2))
PCA <- data.frame(PCA.data_T$x)
PCA_1 <- PCA[,1:10]
PCA_1$ID <- rownames(PCA_1)

 s <- summary(PCA.data_T)
 #str(s)
 PVE <-s$importance[2,] * 100
 #barplot(PVE[1:5], main = "First 5 PCs explain 77% Variation",xlab = "Principal Components",ylab = "Percentage of variances",col ="steelblue", ylim = c(0, 100), space=0.4)
 #lines(x = 1:5, s$importance[3, 1:5]*100, type="b", pch=19, col = "red")
i1 <- round(PVE[1],2)
i2 <- round(PVE[2],2)
i3 <- round(PVE[3],2)
```

Conduct PCA analysis with TMM normalised log2 values to see if there a separation of samples based on Geno Group. PCA analysis was performed on N-11,683 genes. 

:::update
**The PCA plots (first three PCs); PC1 vs PC2 shows that the samples are seperated based on Geno group.** 
:::

```{r, fold.plot=FALSE, fig.width=12,fig.height=12, echo = F}
Geno <- info_sub$Geno
label <- PCA_1$ID
P1 <- ggplot(PCA, aes(x = PC1, y = PC2, colour=Geno)) + geom_point(size=5) +
  theme (axis.text.x  = element_text(size=18), axis.text.y = element_text(size=18)) + 
  theme(axis.title=element_text(size=18,face="bold")) + 
  geom_text(aes(label=(label)),hjust=0.3, vjust=1.5, size = 4) + theme(legend.position='bottom', legend.title = element_text(size = 14),legend.text = element_text(size = 14)) + xlab(paste0("PC1 (", i1, "%)")) + ylab(paste0("PC2 (", i2, "%)")) + ggtitle("A. PC1 vs PC2 - Coloured by Geno Group") + theme(plot.title = element_text(size = 16, face = "bold"))

P2 <- ggplot(PCA, aes(x = PC1, y = PC3, colour=Geno)) + geom_point(size=5) +
  theme (axis.text.x  = element_text(size=18), axis.text.y = element_text(size=18)) + 
  theme(axis.title=element_text(size=18,face="bold")) + 
  geom_text(aes(label=(label)),hjust=0.3, vjust=1.5, size = 4) + theme(legend.position='bottom', legend.title = element_text(size = 14),legend.text = element_text(size = 14)) + xlab(paste0("PC1 (", i1, "%)")) + ylab(paste0("PC3 (", i3, "%)")) + ggtitle("B. PC1 vs PC3 - Coloured by Geno Group") + theme(plot.title = element_text(size = 16, face = "bold"))

P3 <- ggplot(PCA, aes(x = PC2, y = PC3, colour=Geno)) + geom_point(size=5) +
  theme (axis.text.x  = element_text(size=18), axis.text.y = element_text(size=18)) + 
  theme(axis.title=element_text(size=18,face="bold")) + 
  geom_text(aes(label=(label)),hjust=0.3, vjust=1.5, size = 4) + theme(legend.position='bottom', legend.title = element_text(size = 14),legend.text = element_text(size = 14)) + xlab(paste0("PC2 (", i2, "%)")) + ylab(paste0("PC3 (", i3, "%)")) + ggtitle("C. PC2 vs PC3 - Coloured by Geno Group") + theme(plot.title = element_text(size = 16, face = "bold"))

grid.arrange(P1, P2, P3, nrow=2)

```

### PC1vsRNA.Conc

Seems like a slight RNA concentration effect - As all samples with RNA Concentration > 140 are from Geno D group

```{r, fold.plot=FALSE, fig.width=5,fig.height=4, echo = F}
PCA2 <- merge(PCA_1, info_sub, by.x= "ID", by.y="Mouse")
ggplot(PCA2, aes(x = PC1, y = RNA.conc, colour = Geno)) + 
  geom_point(size=4)  +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("RNA Concentration") + ggtitle("PC1 vs RNA Concentration - Coloured by Geno") +  theme(plot.title = element_text(size = 12)) + geom_text(aes(label=(ID)),hjust=0.4, vjust=1.8, size = 3)
```

### PC1vsReadDepth

No obvious read depth effect

```{r, fold.plot=FALSE, fig.width=5,fig.height=4, echo = F}
PCA2 <- merge(PCA_1, info_sub, by.x= "ID", by.y="Mouse")
PCA2$RawReadDepth <- PCA2$Seq.Reads/1000000
ggplot(PCA2, aes(x = PC1, y = RawReadDepth, colour = Geno)) + 
  geom_point(size=4) +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("Million Reads") +ggtitle("PC1 vs Raw Read Depth - Coloured by Geno") + theme(plot.title = element_text(size = 12)) + geom_text(aes(label=(ID)),hjust=0.4, vjust=1.8, size = 3)
```

### PC1vPC3

PC3 somewhat seperates samples in two groups - checking is these groups have difference in % of optical duplicates  
Optical duplicates were binned into two groups >12% and < 12% 
- No clear seperation of samples based on optical duplicates 

```{r, fold.plot=FALSE, fig.width=5,fig.height=4, echo = F}
PCA3 <- merge(PCA2, opts, by.x= "ID", by.y="Mouse")
PCA3$opts <- PCA3$PercentOptical
PCA3$opts[which(PCA3$PercentOptical > 12)] <- ">12%"
PCA3$opts[which(PCA3$PercentOptical < 12)] <- "<12%"
ggplot(PCA3, aes(x = PC1, y = PC2, colour = opts)) + 
  geom_point(size=4) + scale_colour_manual(values=c("#C77CFF","#0d5902", "#f53224", "black")) +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("PC3") +ggtitle("PC1 vs PC3 - Coloured by Optical Duplicate Groups") + theme(plot.title = element_text(size = 12)) + geom_text(aes(label=(ID)),hjust=0.4, vjust=1.8, size = 3)
```

# DE Analysis

Differential analysis was performed using DESeq2

## DESeq2

The pie chart shows the number of up- and down-regulated genes in A vs D 

```{r warning=FALSE, message = FALSE, fig.width=4,fig.height=4, echo = F, warning=FALSE}

rownames(d$counts) <- d$genes$Geneid
cts <- d$counts
coldata <- info_sub[, c("Mouse", "Geno")]
coldata$Mouse <- as.character(coldata$Mouse)
coldata$Geno <- as.character(coldata$Geno)
coldata$condition <- as.factor(gsub("-", "_", coldata$Geno))
coldata$condition <- as.character(coldata$condition)
coldata$condition <- as.factor(coldata$condition)
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~condition)
dds <- DESeq(dds)
DE_out <- results(dds, contrast=c("condition", "A", "D"))
#SRC added this line to export raw csv
write.csv(DE_out, file="deseq2_out_DvsA.csv")
DE_out1 <- DE_out[which(DE_out$padj < 0.05),]
DE_out1 <- DE_out1[order(DE_out1$padj),]
DE_out2 <- DE_out1[,c("log2FoldChange", "pvalue", "padj")]
DE_out2 <- data.frame(DE_out2)
colnames(DE_out2) <- c("Log2FC", "Pvalue", "FDR")
DE_out2$Log2FC <- round(DE_out2$Log2FC, 2)
DE_out2$Pvalue <- format(DE_out2$Pvalue,digits=3)
DE_out2$FDR <- format(DE_out2$FDR,digits=3)


Reg <- c("Up", "Down")
Num <- c(paste0(sum(DE_out2$Log2FC > 0)," UpReg."), paste0(sum(DE_out2$Log2FC < 0)," DownReg."))
df <- data.frame(Reg, Num)
ggplot(df, aes(x = "", y = Num, fill = Reg)) +
  geom_col(color = "black") +
  geom_text(aes(label = Num),
            position = position_stack(vjust = 0.5), size=4) +
  coord_polar(theta = "y") + theme_classic() + ylab("") +
   xlab("") +
   labs(fill="") +
  theme(axis.line = element_blank(),
          axis.text = element_blank(),
          axis.ticks = element_blank(),
         panel.grid  = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x = element_blank()) +
  scale_fill_manual(values=c("#9ecae1", "#fc9272")) + guides(fill=FALSE)

```

The table shows the top 20 genes DE between A vs D genotypes

```{r warning=FALSE, message = FALSE }
kable(DE_out2[1:20,], caption = "A_vs_D") %>%
  kable_styling(bootstrap_options = c("striped", "hover", "responsive"), full_width = F, font_size = 13, fixed_thead = T) %>%
scroll_box(width = "73%", height = "300px")
```

## Sig.DEGenes

The plots show the normalised expression (obtained from DESeq2) of top three genes across the two Geno Groups

```{r, fold.plot=FALSE, fig.width=15,fig.height=5, echo = F, warning=FALSE}
norm <- plotCounts(dds, gene="Mthfd2", intgroup="Geno", returnData=TRUE)
p1 <- ggplot(norm, aes(x = Geno, y = count, color = Geno)) + geom_point(position=position_jitter(w = 0.1,h = 0)) +geom_text_repel(aes(label = rownames(norm))) + 
theme_bw() + ggtitle("Mthfd2") + theme(plot.title = element_text(hjust = 0.5)) + theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("Normalised Counts")

norm <- plotCounts(dds, gene="Slc7a5", intgroup="Geno", returnData=TRUE)
p2 <- ggplot(norm, aes(x = Geno, y = count, color = Geno)) + geom_point(position=position_jitter(w = 0.1,h = 0)) +geom_text_repel(aes(label = rownames(norm))) + 
theme_bw() + ggtitle("Slc7a5") + theme(plot.title = element_text(hjust = 0.5)) + theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("Normalised Counts")

norm <- plotCounts(dds, gene="Psat1", intgroup="Geno", returnData=TRUE)
p3 <- ggplot(norm, aes(x = Geno, y = count, color = Geno)) + geom_point(position=position_jitter(w = 0.1,h = 0)) +geom_text_repel(aes(label = rownames(norm))) + 
theme_bw() + ggtitle("Psat1") + theme(plot.title = element_text(hjust = 0.5)) + theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12),axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12)) + ylab("Normalised Counts")

grid.arrange(p1, p2, p3, nrow=1)
```

## Heatmap 
Heatmap of all the DE Gene.Rows were scaled. The TMM Normalised-log2CPM gene expression values 
were used for the heatmap

```{r, fold.plot=FALSE, fig.width=6,fig.height=7, echo = F, warning=FALSE}
### Annotate our heatmap (optional)
annotation <- info_sub[,c("Mouse", "Geno")]
rownames(annotation) <- annotation$Mouse

### Set a color palette
heat_colors <- rev(brewer.pal(11,"RdBu"))

### Run pheatmap
rownames(datNorm) <- d$genes$Geneid
datNorm1 <- datNorm[which(rownames(datNorm) %in% rownames(DE_out2)),]
pheatmap(datNorm1, 
         color = heat_colors, 
         cluster_rows = T, 
         show_rownames = F,
         annotation = annotation, 
         border_color = NA, 
         fontsize = 10, 
         scale = "row", 
         fontsize_row = 10, 
         height = 20)
```

## VolcanoPlots

Note- for visualisation purpose -log10FDR values > 50 were replaced with 50

```{r, fold.plot=FALSE, fig.width=5,fig.height=5, echo = F, warning=FALSE}
v <- data.frame(DE_out[, c("log2FoldChange", "padj")])
v$Gene <- rownames(v)

##Highlight genes with color
v$logFC <-v$log2FoldChange 
v$FDR <- v$padj
v$Col <- rep("No", nrow(v))
v$Col[which(v$FDR < 0.05 & v$logFC >= -1)] <- "darkgreen"
v$Col[which(v$FDR < 0.05 & v$logFC > 0 & v$logFC <= 1)] <- "red"
v$Col[which(v$FDR< 0.05 & v$logFC > 1)] <- "red"
v$Col[which(v$FDR < 0.05 & v$logFC < -1)] <- "darkgreen"
v <- v[-which(is.na(v$FDR)),]
v$FDR1 <- -log10(v$FDR)
v$FDR1[which(v$FDR1 > 50)] <- 50
a <- which(v$FDR1 > 20)
b <- which(v$FDR1 > 8 & v$log2FoldChange < -1)
c <- c(a,b)
v1 <- v[c,]
#pdf(paste0(outfile_path, "VolcanoPlots/VolcanoPlot_STZNULL_vs_CITNULL.pdf"), width=6, height = 5)
g = ggplot(data=v, aes(x=logFC, y=FDR1, colour= Col)) + geom_point(alpha=0.9, size=3, show.legend = FALSE)  + xlim(c(-2.5, 5)) + scale_colour_manual(values=c("blue", "grey35", "firebrick2")) +
xlab("log2 fold change") + ylab("-log10 FDR") + ggtitle("A_vs_D")
h <- g + theme_linedraw() + theme_bw() + theme(axis.text.x = element_text(color="black", size=14), axis.text.y = element_text(color="black", size=14), axis.title.x = element_text(color="black", size=12, face="bold"), axis.title.y = element_text(face="bold", size=12))
v$Abs <- abs(as.numeric(as.character(v$logFC)))
h + geom_text(data=v1, angle = 45, size = 2.7, hjust = 0.5, vjust = -0.5, colour = "black", aes(logFC,FDR1, label=Gene)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
panel.background = element_blank(), axis.line = element_line(colour = "black")) 
```

